### PR TITLE
Update user guide for service account changes

### PR DESCRIPTION
**Pull request by idlewis**
Docs for issue:
https://github.com/OpenLiberty/open-liberty-operator/issues/462

### CODE REVIEW

File: doc/user-guide-v1.adoc [hunk 1]
```diff
@@ -156,7 +156,10 @@ Each `OpenLibertyApplication` CR must specify `.spec.applicationImage` field. Sp
...
 | `serviceability.volumeClaimName` | [[crd-spec-serviceability-volumeClaimName]] The name of the link:++https://kubernetes.io/docs/concepts/storage/persistent-volumes/#persistentvolumeclaims++[PersistentVolumeClaim] resource you created to be used for serviceability. Must be in the same namespace.
-| `serviceAccountName` | The name of the Red Hat OpenShift service account to be used during deployment. If a service account name is not specified, a service account is automatically created. For examples, see link:#create-a-service-account[Create a service account].
+| `serviceAccountName` | Deprecated. Use link:#crd-spec-serviceAccount-name[`serviceAccount.name`] instead.
+| `serviceAccount`| [[crd-spec-serviceAccount-name]] The name of the service account to be used during deployment. If a service account name is not specified, a service account is automatically created. For examples, see link:#create-a-service-account[Create a service account].
```
Comment (leochr) [inline doc/user-guide-v1.adoc:160]: Suggestion (match the WLO doc):
`The service account to use for application deployment. If a service account name is not specified, a service account is automatically created. For examples, see link:#create-a-service-account[Create a service account].`

**Review by leochr — COMMENTED** [review]
@idlewis Thanks for the doc updates. Added a couple of minor comments.

File: doc/user-guide-v1.adoc [hunk 2]
```diff
@@ -333,7 +336,7 @@ Open Liberty Operator builds upon link:#common-component-documentation[component
 ==== Common Components [[common-component-documentation]]
 
 * link:#reference-image-streams[Reference image streams] (`.spec.applicationImage`)
-* link:#create-a-service-account[Create a service account] (`.spec.serviceAccountName`)
+* link:#create-a-service-account[Create a service account] (`.spec.serviceAccount.name`)
```
[the remark this answers is not in the document]
Comment (leochr) [reply doc/user-guide-v1.adoc:339]: Sorry, I meant to suggest the following for title: `Configure service account (.spec.serviceAccount)`

**Comment by idlewis** [comment]
@leochr I've addressed your comments I think, could you take another look?

**Review by leochr — APPROVED** [review]
@idlewis Looks good. Thank you

**Pull request merged**